### PR TITLE
feat(new construct): aws-lambda-kinesisfirehose

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*
Closes Issue #874 

*Description of changes:*
Allows the creation of a Lambda function to an existing Kinesis Firehose construct. This allows architects to set up a firehose with multiple sources. It also sets us up to have sources for new firehose destinations on the roadmap.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 853ff276e635c90e09e260acb27cb51f388900b1
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4742c01b-67db-41b2-b6d7-ed481b7cc698%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# aws-lambda-s3 module
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md:1]: update to `aws-lambda-kinesisfirehose`

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Out of the box implementation of the Construct without any override will set the following defaults:
+
+### AWS Lambda Function
+* Configure limited privilege access IAM role for Lambda function
+* Enable reusing connections with Keep-Alive for NodeJs Lambda function
+* Enable X-Ray Tracing
+* Set Environment Variables
+  * (default) FIREHOSE_DELIVERYSTREAM_NAME
+  * AWS_NODEJS_CONNECTION_REUSE_ENABLED
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md:109]: Just an observation, the default behavior for this flag changed in later versions of the node lambda runtime. (defaults to true in 18). It also feels weird to have this environment variable that is specific to node, when the lambda could be any language?

**Review by georgebearden — CHANGES_REQUESTED** [review]
Update the architecture diagram also!

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 735e13920e540a8b4e8863b712a97482b7e68569
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=83efcc39-4a3a-4402-9545-3d8ec6c6ae57%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by georgebearden — APPROVED** [review]
Everything looks good to me. Should we update the architecture diagram kinesis component to say `Amazon Kinesis Firehose Delivery Stream` (the Firehose part is missing)?

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: dbcf8ff6835adde49503a9566aa5fe997071d480
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8462d33a-d002-4122-b053-7b0840e1b317%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 2dee56ccc3bc2172c926c2a6fad023d2251536f2
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=45570b84-5e76-44e8-973d-264f2bbaa04e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 6b81b069c9e471c604898c42258349f7633cc5bc
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6761791e-2fd6-4168-8d24-336e436f301c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f063254b64a4baf1089d6ad882b2cb14f79f37e4
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ecf54bbb-e631-4b32-9291-85d1851eb0b2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: a7d85f5e0fb434fdd194d80b15f7fa083c70be70
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9dcaead0-3dc7-43ce-8f29-e581f53c1cf3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 1ff8adff94866fb6b96301a505d81043df7768bb
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=34c99714-130c-443f-b14b-b932ce58a691%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f7f37d213f0093ba95075fc9dec35d12512e6383
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2c37055e-58d4-4a58-80c5-005aa3ed9603%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 1ceaed12e220f69e2fc4056776e63421d42a72d5
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4a81c39b-3ab1-4ef5-993e-25a2531b0dee%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/lib/index.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    this.lambdaFunction = defaults.buildLambdaFunction(this, {
+      existingLambdaObj: props.existingLambdaObj,
+      lambdaFunctionProps: props.lambdaFunctionProps,
+      vpc: this.vpc,
+    });
+
+    // Setup the firehose
+    this.kinesisFirehose = props.existingKinesisFirehose;
+
+    if (!this.kinesisFirehose.deliveryStreamName) {
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/lib/index.ts:112]: I see we follow a pattern of when we provision resources in the index files, but should this check go up closer to the `checkProps`, as everything else done up to this point will be invalid if this fails?

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+|lambdaFunction|[`lambda.Function`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html)|Returns an instance of the Lambda function created by the pattern.|
+|kinesisFirehose|[kinesisfirehose.CfnDeliveryStream](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kinesisfirehose.CfnDeliveryStream.html)|The Kinesis Firehose Delivery Stream used by the construct.|
+|vpc?|[`ec2.IVpc`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.IVpc.html)|Returns an interface on the VPC used by the pattern (if any). This may be a VPC created by the pattern or the VPC supplied to the pattern constructor.|
+
+## Default settings
+
+Out of the box implementation of the Construct without any override will set the following defaults:
+
+### AWS Lambda Function
+* Configure limited privilege access IAM role for Lambda function
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/README.md:104]: Might be nice to say ` Configure limited privilege access IAM role for Lambda function with permission to put records on the firehose`

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/lib/index.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  public readonly lambdaFunction: lambda.Function;
+  public readonly vpc?: ec2.IVpc;
+  public readonly kinesisFirehose: firehose.CfnDeliveryStream;
+
+  /**
+   * @summary Constructs a new instance of the LambdaToKinesisFirehose class.
+   * @param {cdk.App} scope - represents the scope for all the resources.
+   * @param {string} id - this is a a scope-unique id.
+   * @param {LambdaToKinesisFirehoseProps} props - user provided props for the construct.
+   * @since 0.8.0
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/lib/index.ts:81]: I've glossed over this in previous constructs I've worked on, but what is the significance of this `@since 0.8.0`?

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/package.json [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    }
+  },
+  "dependencies": {
+    "@aws-cdk/aws-lambda": "0.0.0",
+    "@aws-cdk/aws-kinesisfirehose": "0.0.0",
+    "@aws-cdk/aws-ec2": "0.0.0",
+    "@aws-cdk/core": "0.0.0",
+    "@aws-solutions-constructs/core": "0.0.0",
+    "@aws-solutions-constructs/aws-kinesisfirehose-s3": "0.0.0",
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/package.json:61]: Is this an accurate dependency?

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    }
+  });
+
+  expect(construct.kinesisFirehose).toBeDefined();
+  expect(construct.lambdaFunction).toBeDefined();
+  expect(construct.vpc).not.toBeDefined();
+});
+
+test('Test existing function is utilzed correctly', () => {
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts:42]: Nit spelling - `utilzed` -> `utilized`

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+    existingKinesisFirehose: destination.kinesisFirehose,
+    existingLambdaObj: existingFunction
+  });
+
+  expect(stack).toHaveResource('AWS::Lambda::Function', {
+    FunctionName: testName
+  });
+
+  // This is 2 because there's a lambda function in the custom resource to
+  // delete all the objects when cleaning up the s3 bucket in kinesisfirehose-s3
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts:64]: In going through the PR top-down, I'm surprised to see references to kinesisfirehose-s3. Is this comment accurate? Its not obvious to me why/how we are using it.

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/package.json (hunk 6 above)
Comment (georgebearden) [reply source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/package.json:61]: Now I see it in the tests. Perhaps this should at least be only a `devDependency` then? Also - no issue with a construct depending on another construct like this? I guess we have the higher-level L3/4? constructs that do this all the time.

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts (hunk 8 above)
Comment (georgebearden) [reply source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts:64]: Now I see why below in the definition of the `GetTestDestination` function.

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+  const destination = GetTestDestination(stack, 'test-destination');
+
+  const construct = new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    },
+    deployVpc: true,
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts:103]: What happens if deployVpc is false or undefined, and the user still includes `vpcProps`? Do we expect the construct to still use them, or do we throw an error?

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+});
+
+test('Test fail if existingFirehose does not have a stream name', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+  const testName = 'test-name';
+
+  const mutatedFirehose = defaults.overrideProps(destination.kinesisFirehose, {});
+  mutatedFirehose.deliveryStreamName = undefined;
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts:180]: nice!

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/integ.existing-function.ts [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import {LambdaToKinesisFirehose } from "../lib";
+import { generateIntegStackName } from '@aws-solutions-constructs/core';
+import { KinesisFirehoseToS3 } from '@aws-solutions-constructs/aws-kinesisfirehose-s3';
+import * as defaults from '@aws-solutions-constructs/core';
+
+// Setup
+const app = new App();
+const stack = new Stack(app, generateIntegStackName(__filename));
+stack.templateOptions.description = 'Integration Test for aws-cdk-apl-kinesisfirehose-s3';
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/integ.existing-function.ts:25]: Nit - is this description accurate? I'm not sure what apl is.

Comment (georgebearden) [reply source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/integ.existing-function.ts:25]: Same question in the other integ test stack descriptions.

File: source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/lambda/index.mjs [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import { FirehoseClient, PutRecordCommand } from "@aws-sdk/client-firehose";
+const client = new FirehoseClient({});
+
+export const handler = async(event) => {
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/lambda/index.mjs:4]: Nice test that auto puts records on the stream! Makes the validation a whole lot easier having the test lambda function do it for us!

File: source/patterns/@aws-solutions-constructs/core/lib/utils.ts [hunk 13]
```diff
@@ -176,4 +177,20 @@ export function consolidateProps(defaultProps: object, clientProps?: object, con
...
+/**
+ * Generates a name unique to this location in this stack with this stackname. Truncates to under 64 characters if needed.
+ * (will allow 2 copies of the stack with different stack names, but will collide if both stacks have the same name)
+ *
+ * @param scope the construct within to create the name
+ * @param resourceId an id for the construct about to be created under scope (empty string if name is for scoep)
+ * @returns a unique name
+ */
+export function generateName(scope: Construct, resourceId: string = ""): string {
+  const name = /*cdk.Stack.of(scope).region +*/ resourceId + cdk.Names.uniqueId(scope);
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/core/lib/utils.ts:191]: is the commented out `cdk.Stack.of(scope).region` intentional? If so, can we add a comment as to what the intent of it is?

File: source/patterns/@aws-solutions-constructs/core/test/utils.test.ts [hunk 14]
```diff
@@ -142,4 +143,42 @@ test('Test consolidate props with one arg', () => {
 
   expect(result).toEqual(arg1);
 
-});
\ No newline at end of file
+});
+
+test('Test generateName sunny day for current construct with unefined name argument', () => {
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/core/test/utils.test.ts:148]: Nit spelling - `unefined` -> `undefined`

File: source/patterns/@aws-solutions-constructs/core/test/utils.test.ts [hunk 15]
```diff
@@ -142,4 +143,42 @@ test('Test consolidate props with one arg', () => {
...
+  const stack = new cdk.Stack(undefined, "some-new-id");
+
+  const newName = defaults.generateName(stack);
+  expect(newName.length).toBeGreaterThan(5);
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/core/test/utils.test.ts:152]: Just so I understand, why is the number 5 significant here?

**Review by georgebearden — CHANGES_REQUESTED** [review]
Nice Work! Mostly minor nits/questions - one larger concern around an edge case when the existing firehose is encrypted with a CMK (comment is in the index.ts file).

File: source/patterns/@aws-solutions-constructs/core/test/utils.test.ts (hunk 15 above)
Comment (biffgaut) [reply source/patterns/@aws-solutions-constructs/core/test/utils.test.ts:152]: It's a bit random - just making that it is set to something and not get fooled by a newline or something like that

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: a80c299c940d679d92c104c921c2d9558ef4e10b
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=92b33c0e-6630-4b2a-b579-18ff3cc0187d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 4e4d2c6cc4021fcd801e568a278ad484407a1576
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf0c9d52-4a2c-4a58-8ef5-12cb1897bb5f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d3b574beb0de2a1f38dd0d3627d282f140b4e69b
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ee52fe17-d52c-4dcd-b2fe-c1e4e8af5f3c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: ab9299f5768fb5f5057654a9f5803c9d8c9db8a4
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ce94b4ca-1c4f-4a49-8976-a9e432a98e36%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: d7807634f2b85443764cbd69f6376e500fbd9cb1
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf096170-c3d2-4afe-865b-7d2a6cd78c1e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: fec37e86c6b56e0cb503feacc586142b839139f7
* Result: FAILED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9753094b-682d-4007-a6d7-8f6ee28b0b20%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f923d8b8fcc06fccfd0e97aab3ed9b056330fdff
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=29adeb01-f155-4f5e-b5b7-fc6a50053eec%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by georgebearden — APPROVED** [review]
looks good to me!
